src/emailmessage.cpp

```diff
@@ -17,6 +17,25 @@
 #include "qmailnamespace.h"
 #include <qmaildisconnected.h>
 #include <QTextDocument>
+#include <QTemporaryFile>
+#include <QStandardPaths>
+#include <QDir>
+
+// Part finder
+struct PartFinder {
+    PartFinder(QMailMessageContentType type, const QMailMessagePart *&part) : contentType(type), partFound(part) {}
```

Comment:
The main difference is, you don't need const-ref-to-mutable-pointer in the interface :-)
The value is copied out immediately prior destruction in the only place you currently use this object, so going out of scope isn't a problem.